packages/pds/tests/db-notify.test.ts

```diff
@@ -0,0 +1,67 @@
+import { wait } from '@atproto/common'
+import { Database } from '../src'
+
+describe('db', () => {
+  let dbOne: Database
+  let dbTwo: Database
+
+  beforeAll(async () => {
+    if (process.env.DB_POSTGRES_URL) {
+      dbOne = Database.postgres({
+        url: process.env.DB_POSTGRES_URL,
+        schema: 'db_notify',
+      })
+      dbTwo = Database.postgres({
+        url: process.env.DB_POSTGRES_URL,
+        schema: 'db_notify',
+      })
+    } else {
+      dbOne = Database.memory()
+      dbTwo = dbOne
+    }
+  })
+
+  afterAll(async () => {
+    await dbOne.close()
+    await dbTwo.close()
+  })
+
+  it('notifies', async () => {
+    const toSend = ['one', 'two', 'three', 'four', 'five']
+    const received: string[] = []
+    await dbOne.listenFor('test', (msg) => {
+      received.push(msg || '')
+    })
+
+    dbTwo.notify('otherchannel', 'blah')
+    for (const msg of toSend) {
+      dbTwo.notify('test', msg)
+    }
+    dbTwo.notify('otherchannel', 'blah')
+
+    await wait(200)
```

Comment:
don't love just throwing in an arbitrary wait here, but not sure what else to do 🤷

Reply:
Has a different drawback, where if the test is failing it could hang, but one way to slice this is with a little home-spun deferrable.

```ts
    const deferrables = [createDeferrable(), createDeferrable()]
    await dbOne.listenFor('test', (msg) => {
      deferrables[received.length].resolve()
      received.push(msg || '')
    })
    // ...
    await Promise.all(deferrables.map((d) => d.complete))

// ...
const createDeferrable = () => {
  let resolve
  const promise = new Promise((res) => {
    resolve = res
  })
  return { resolve, complete: promise }
}
```

Reply:
yeah nice this feels better 👍